firebase-storage/src/main/java/com/google/firebase/storage/StorageReference.java

```diff
@@ -563,14 +566,119 @@ public StreamDownloadTask getStream(@NonNull StreamDownloadTask.StreamProcessor
   /**
    * Deletes the object at this {@link StorageReference}.
    *
-   * @return a {@link Task} that indicates whether the operation succeeded or failed.
+   * @return A {@link Task} that indicates whether the operation succeeded or failed.
    */
+  @NonNull
   @PublicApi
   public Task<Void> delete() {
     TaskCompletionSource<Void> pendingResult = new TaskCompletionSource<>();
     StorageTaskScheduler.getInstance().scheduleCommand(new DeleteStorageTask(this, pendingResult));
     return pendingResult.getTask();
   }
+
+  /**
+   * List up to `maxResults` items (files) and prefixes (folders) under this StorageReference.
+   *
+   * <p>"/" is treated as a path delimiter. Firebase Storage does not support invalid object paths
+   * that end with "/" or contain two consecutive "//". All invalid objects in GCS will be filtered.
```

Comment:
```suggestion
   * that end with "/" or contain two consecutive "/"s. All invalid objects in GCS will be filtered.
```
Maybe mention if there are too many `unsupported` objects in GCS, list may always fail.

Consider mirror the doc from JS SDK PR to stay consistent.

Reply:
I do slightly prefer having smaller docstrings. Ideally, the backend would throw an error that is explanatory enough if this edge case occurs.